apps/src/code-studio/pd/workshop_dashboard/reports/teacher_progress_report.jsx

```diff
@@ -46,19 +47,21 @@ const TeacherProgressReport = React.createClass({
     if (
       nextProps.startDate !== this.props.startDate ||
       nextProps.endDate !== this.props.endDate ||
-      nextProps.queryBy !== this.props.queryBy
+      nextProps.queryBy !== this.props.queryBy ||
+      nextProps.course !== this.props.course
     ) {
-      this.load();
+      this.load(nextProps);
```

Comment:
It's worth noting that there could be a race condition here, because you're calling `load` again without checking for and killing the previous `loadRequest`. The fact that you are hiding the controls while the request is in progress may prevent a race from happening in practice, but depending on what these results are used for it might be worth checking `loadRequst` anyway to be sure.